### PR TITLE
Emit events when replaying external txns

### PR DESCRIPTION
**Pull request by pmconne**
See https://github.com/iTwin/itwinjs-core/pull/5867
itwinjs-core: https://github.com/iTwin/itwinjs-core/tree/pmc/external-txn-events
imodel-native-internal: https://github.com/iTwin/imodel-native-internal/tree/pmc/external-txn-events

TxnManager::ModelChanges seemed to assume changes can't be applied to a read-only briefcase. That's certainly not the case when the `watchForChanges` flag is enabled when opening the briefcase.
As a result, it would fail to update a model's range index when its geometry changed.

### CODE REVIEW

**Comment by pmconne** [comment]
@swbsi or @tcobbs-bentley can you please submit a PR to fix or reduce the android build failures?
```
Execution failed for task ':app:mergeReleaseNativeDebugMetadata'.
> A failure occurred while executing com.android.build.gradle.internal.tasks.MergeNativeDebugMetadataTask$MergeNativeDebugMetadataWorkAction
   > Java heap space
```

**Comment by tcobbs-bentley** [comment]
> @swbsi or @tcobbs-bentley can you please submit a PR to fix or reduce the android build failures?

It looks like it's already set to use 4GB. We could of course bump that up, but I don't know if the error is because it's going over 4GB or if 4GB is not available. @swbsi, what do you think?

Note that on my Mac, I have never had an Android build fail when Gradle is configured to use 4GB.

**Comment by swbsi** [comment]
> > @swbsi or @tcobbs-bentley can you please submit a PR to fix or reduce the android build failures?
> 
> It looks like it's already set to use 4GB. We could of course bump that up, but I don't know if the error is because it's going over 4GB or if 4GB is not available. @swbsi, what do you think?
> 
> Note that on my Mac, I have never had an Android build fail when Gradle is configured to use 4GB.

I have also never had an issue at 4gb locally.

Not sure if the error is really a system oom situation…maybe try a really large value locally (that you know will exceed physical+virtual on your box) and compare the resulting error?

**Comment by tcobbs-bentley** [comment]
> Not sure if the error is really a system oom situation…maybe try a really large value locally (that you know will exceed physical+virtual on your box) and compare the resulting error?

I'm not sure I'm willing to try doing that. What if we change it from 4GB to 3GB?

**Comment by pmconne** [comment]
@swbsi @tcobbs-bentley [here](https://github.com/iTwin/imodel-native/issues/418) is a better place to discuss and track.

**Review by nick4598 — APPROVED** [review]

File: iModelCore/iModelPlatform/DgnCore/DgnModel.cpp [hunk 1]
```diff
@@ -821,7 +821,8 @@ BeGuid GeometricModel::QueryGeometryGuid() const {
     stmt.BindId(1, m_modelId);
     stmt.Step();
 
-    return stmt.GetValueGuid(0);
+    auto guid = stmt.GetValueGuid(0);
```
Comment (tcobbs-bentley) [inline iModelCore/iModelPlatform/DgnCore/DgnModel.cpp:824]: This looks like a change that was made for debugging purposes. I don't mind leaving it in if desired, but you may not have intended to commit it.

**Review by khanaffan — APPROVED** [review]

File: iModelCore/iModelPlatform/DgnCore/DgnModel.cpp (hunk 1 above)
Comment (pmconne) [reply iModelCore/iModelPlatform/DgnCore/DgnModel.cpp]: Reverted.

**Comment by pmconne** [comment]
@DanRod1999 @aruniverse is this supposed to show my linked PR branches?

image

**Comment by DanRod1999** [comment]
> @DanRod1999 @aruniverse is this supposed to show my linked PR branches?

No. This only shows the initial branch that is pulled in by the pipeline. Your linked branch would be shown later inside the actual pipeline run on imodel-native-internal here:
image

**Pull request merged**